### PR TITLE
bugfix/url params names

### PR DESCRIPTION
**Pull request by baywet**
related 
https://github.com/microsoft/kiota/issues/1445

https://github.com/microsoft/kiota-abstractions-dotnet/pull/11

### CODE REVIEW

File: src/Middleware/ParametersNameDecodingHandler.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    /// </summary>
+    /// <param name="options">An OPTIONAL <see cref="ParametersNameDecodingOption"/> to configure <see cref="ParametersNameDecodingHandler"/></param>
+    public ParametersNameDecodingHandler(ParametersNameDecodingOption options = default)
+    {
+        EncodingOptions = options ?? new();
+    }
+    ///<inheritdoc/>
+    protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage httpRequestMessage, CancellationToken cancellationToken)
+    {
+        if(!httpRequestMessage.RequestUri.Query.Contains('%') ||
```
Comment (andrueastman) [inline src/Middleware/ParametersNameDecodingHandler.cs:34]: We should probably check if there is a `ParametersNameDecodingOption` set in the request message itself first to use if it exists to allow custom options on a "per request basis" like the other middleware. 

```cs
EncodingOptions = httpRequestMessage.GetRequestOption<ParametersNameDecodingOption>() ?? EncodingOptions;
```

File: src/Middleware/ParametersNameDecodingHandler.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+/// <summary>
+/// This handlers decodes special characters in the request query parameters that had to be encoded due to RFC 6570 restrictions names before executing the request.
+/// </summary>
+public class ParametersNameDecodingHandler: DelegatingHandler
+{
+    /// <summary>
+    /// The options to use when decoding parameters names in URLs
+    /// </summary>
+    public ParametersNameDecodingOption EncodingOptions { get; set; }
```
Comment (andrueastman) [inline src/Middleware/ParametersNameDecodingHandler.cs:22]: maybe private or internal?
It is ideally set through the constructor and there's the potential path through the request options of the message itself.

File: src/Middleware/ParametersNameDecodingHandler.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+
+/// <summary>
+/// This handlers decodes special characters in the request query parameters that had to be encoded due to RFC 6570 restrictions names before executing the request.
+/// </summary>
+public class ParametersNameDecodingHandler: DelegatingHandler
```
Comment (andrueastman) [inline src/Middleware/ParametersNameDecodingHandler.cs:17]: Should we add this to the default Kiota Client factory?

https://github.com/microsoft/kiota-http-dotnet/blob/8f8f759b21817b34de141adf85adb295f2c0e345/src/KiotaClientFactory.cs#L39

File: src/Middleware/ParametersNameDecodingHandler.cs (hunk 2 above)
Comment (baywet) [reply src/Middleware/ParametersNameDecodingHandler.cs:22]:
```suggestion
    internal ParametersNameDecodingOption EncodingOptions { get; set; }
```

File: src/Middleware/ParametersNameDecodingHandler.cs (hunk 3 above)
Comment (baywet) [reply src/Middleware/ParametersNameDecodingHandler.cs:17]: thanks for the callout, done

File: src/Middleware/ParametersNameDecodingHandler.cs (hunk 1 above)
Comment (baywet) [reply src/Middleware/ParametersNameDecodingHandler.cs:34]: fixed.

**Review by andrueastman — APPROVED** [review]

**Pull request merged**